src/grpc/router_ics_skf_worker.erl

```diff
@@ -131,67 +175,59 @@ handle_cast({?RECONCILE_START, Options}, #state{conn_backoff = Backoff0} = State
-    ]),
-    {noreply, State#state{reconciling = false}};
+    lager:info(
+        "DRY RUN, reconciling done adding ~w removing ~w",
+        [erlang:length(ToAdd), erlang:length(ToRemove)]
+    ),
+    {noreply, done_reconciling(State)};
 handle_cast(
-    {?RECONCILE_END, #{forward_pid := Pid, commit := true} = Options, SKFs},
+    {?RECONCILE_END, #{commit := true} = Options, SKFs},
     #state{oui = OUI} = State
-) when is_pid(Pid) ->
+) ->
     lager:info("got RECONCILE_END ~p, with ~w", [Options, erlang:length(SKFs)]),
-    LocalSKFs = get_local_skfs(OUI),
-    ToAdd = LocalSKFs -- SKFs,
-    ToRemove = SKFs -- LocalSKFs,
-    ok = forward_reconcile(Options, {ok, ToAdd, ToRemove}),
-    lager:info("reconciling done adding ~w removing ~w", [
-        erlang:length(ToAdd), erlang:length(ToRemove)
-    ]),
+    {ToAdd, ToRemove} = local_skf_to_remote_diff(OUI, SKFs),
+    AddCount = erlang:length(ToAdd),
+    RemCount = erlang:length(ToRemove),
+    lager:info("diff with local adding ~w removing ~w", [AddCount, RemCount]),
     case maybe_update_skf([{remove, ToRemove}, {add, ToAdd}], State) of
-        {error, Reason} = Error ->
+        {error, _Reason} = Error ->
             ok = forward_reconcile(Options, Error),
-            {noreply, skf_update_failed(Reason, State#state{reconciling = false})};
+            State1 = maybe_schedule_reconnect(Options#{error => Error}, State),
+            {noreply, done_reconciling(State1)};
         ok ->
             ok = forward_reconcile(Options, {ok, ToAdd, ToRemove}),
-            lager:info("reconciling done adding ~w removing ~w", [
-                erlang:length(ToAdd), erlang:length(ToRemove)
-            ]),
-            {noreply, State#state{reconciling = false}}
-    end;
-handle_cast(
-    {?RECONCILE_END, #{forward_pid := undefined, commit := true} = Options, SKFs},
-    #state{oui = OUI} = State
-) ->
-    lager:info("got RECONCILE_END ~p, with ~w", [Options, erlang:length(SKFs)]),
-    LocalSKFs = get_local_skfs(OUI),
-    ToAdd = LocalSKFs -- SKFs,
-    ToRemove = SKFs -- LocalSKFs,
-    ok = forward_reconcile(Options, {ok, ToAdd, ToRemove}),
-    lager:info("reconciling done adding ~w removing ~w", [
-        erlang:length(ToAdd), erlang:length(ToRemove)
-    ]),
-    case maybe_update_skf([{remove, ToRemove}, {add, ToAdd}], State) of
-        {error, Reason} ->
-            {noreply, skf_update_failed(Reason, State#state{reconciling = false})};
-        ok ->
-            lager:info("reconciling done adding ~w removing ~w", [
-                erlang:length(ToAdd), erlang:length(ToRemove)
-            ]),
-            {noreply, State#state{reconciling = false}}
+            lager:info("reconciling done added ~w removed ~w", [AddCount, RemCount]),
+            {noreply, done_reconciling(State)}
     end;
 handle_cast(
-    {?UPDATE, _Updates}, #state{reconciling = true} = State
+    {?UPDATE, Updates}, #state{reconciling = true, deferred_updates = DeferredUpdates} = State
```

Comment:
This module got the same `reconcile_end` restructuring as the EUI worker.

If an update comes through while we're reconciling, it will get queued. When reconciling is done, we'll process those updates. 
This prevents another entire reconcile that can take a long time and build up more queued updates, or dropping session keys on the floor to never be picked up.